react/bingo/src/bingo/GameLogic.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import './BingoApp.css';
+import { cleanup } from '@testing-library/react';
+
+class GameLogic extends React.Component {
+    constructor(props) {
+        super(props);
+        this.maskedChars = [];
+
+        this.state = {
+            attemptLetter: '',
+            failedLetters: '',
+            successLetters: '',
+            challengeWord: [],
+            maskedWord: [],
+            isGameOver : false,
+            bingoColor: [
+                'attempt-left',
+                'attempt-left',
+                'attempt-left',
+                'attempt-left',
+                'attempt-left'
+            ]
+        };
+
+    }
+    getBlankText = () => {
+        let challengeText = this.props.challenge;
```

Comment:
Can use "Object destructuring" in the constructor to get the values from props assigned to variables, can be used if there are multiple props or if the same props is going to be used in multiple functions within the components.

Reply:
Need to try this out. Leaving the comment unresolved for now.